lib/components/tabs/tabs.dart

```diff
@@ -0,0 +1,136 @@
+import 'dart:html';
+import 'dart:async';
+import 'package:katikati_ui_lib/components/logger.dart';
+
+var logger = Logger('Tabs');
+
+class TabView {
+  String id;
+  String label;
+  DivElement content;
+
+  TabView(this.id, this.label, this.content);
+}
+
+class TabsView {
+  DivElement renderElement;
+  List<TabView> _tabs;
+  Map<String, TabView> _tabsLookup;
+
+  String _selectedTabID;
+  String _tabContentClassname;
+
+  List<SpanElement> _tabChoosers = [];
+  DivElement _tabsHeader;
+  DivElement _tabContent;
+
+  Stream<String> _onEmpty;
+  StreamController<String> _onEmptyController;
+  Stream<String> get onEmpty => _onEmpty;
+
+  TabsView(this._tabs, {String defaultSelectedID, String tabContentClassname}) {
+    _updateLookup();
+
+    if (defaultSelectedID != null) {
+      _selectedTabID = defaultSelectedID;
+    } else {
+      _selectedTabID = _tabs.first.id;
+    }
```

Comment:
Let's make this a one liner:
```suggestion
    _selectedTabID = defaultSelectedID ?? _tabs.first.id;
```
(the `x ?? y` operator is equivalent to `x != null ? x : y)

Also, can we have this constructor throw an error at the beginning if `_tabs` is empty?